metadata-ingestion/src/datahub/ingestion/source/sql_queries.py

```diff
@@ -193,20 +285,154 @@ def get_workunits_internal(
+        else:
+            logger.info("Using batch processing (all queries loaded into memory)")
+            yield from self._process_queries_batch()
+
+        # Log processing statistics
+        self._log_processing_statistics()
+
+    def _process_queries_streaming(
+        self,
+    ) -> Iterable[Union[MetadataWorkUnit, MetadataChangeProposalWrapper]]:
+        """Process queries in streaming fashion to minimize memory usage."""
+        with self.report.new_stage("Streaming query processing"):
+            query_batch = []
+
+            for query_entry in self._parse_query_file():
+                query_batch.append(query_entry)
+
+                # Process batch when it reaches the streaming batch size
+                if len(query_batch) >= self.config.streaming_batch_size:
+                    yield from self._process_query_batch(query_batch)
+                    query_batch = []
+
+            # Process remaining queries in the final batch
+            if query_batch:
+                yield from self._process_query_batch(query_batch)
+
+    def _process_queries_batch(
+        self,
+    ) -> Iterable[Union[MetadataWorkUnit, MetadataChangeProposalWrapper]]:
+        """Process all queries in memory (original behavior)."""
         with self.report.new_stage("Collecting queries from file"):
             queries = list(self._parse_query_file())
             logger.info(f"Collected {len(queries)} queries for processing")
 
         with self.report.new_stage("Processing queries through SQL parsing aggregator"):
-            for query_entry in queries:
-                self._add_query_to_aggregator(query_entry)
+            logger.info("Using sequential processing")
+            self._process_queries_sequential(queries)
 
         with self.report.new_stage("Generating metadata work units"):
             logger.info("Generating workunits from SQL parsing aggregator")
-            yield from self.aggregator.gen_metadata()
+            yield from auto_workunit(self.aggregator.gen_metadata())
 
-    def _parse_query_file(self) -> Iterable["QueryEntry"]:
-        """Parse the query file and yield QueryEntry objects."""
+    def _process_query_batch(
+        self, query_batch: List["QueryEntry"]
+    ) -> Iterable[Union[MetadataWorkUnit, MetadataChangeProposalWrapper]]:
+        """Process a batch of queries and yield metadata work units."""
+        self.report.num_streaming_batches_processed += 1
+        logger.debug(
+            f"Processing streaming batch {self.report.num_streaming_batches_processed} with {len(query_batch)} queries"
+        )
+
+        # Process the batch
+        self._process_queries_sequential(query_batch)
+
+        # Generate and yield metadata work units for this batch
+        # Note: No stage created here to avoid excessive stage creation for each batch
+        yield from auto_workunit(self.aggregator.gen_metadata())
+
+    def _log_processing_statistics(self) -> None:
```

Comment:
All these metrics will be visible in the report already

Logging a textual summary is not a pattern we have done before